microsoft-sentinel/src/openbas_microsoft_sentinel.py

```diff
@@ -122,6 +123,16 @@ def _extract_process_names(self, columns_index, alert):
                 process_names.append(entity["Name"])
         return process_names
 
+    def _extract_parent_process_name(self, columns_index, alert):
+        parent_process_names = []
+        entities = json.loads(alert[columns_index["Entities"]])
+        for entity in entities:
+            if "Type" in entity and entity["Type"] == "process":
+                if "ParentProcess" in entity and "ImageFile" in entity["ParentProcess"]:
+                    if "ImageFile" in entity["ParentProcess"] and "Name" in entity["ParentProcess"]["ImageFile"]:
+                        parent_process_names.append(entity["ParentProcess"]["ImageFile"]["Name"])
+        return parent_process_names
```

Comment:
**suggestion (non-blocking):** ifs nesting could be removed/improved

```
def _extract_parent_process_name(self, columns_index, alert):
    parent_process_names = []
    entities = json.loads(alert[columns_index["Entities"]])
    
    for entity in entities:
        if (entity.get("Type") == "process" and
            entity.get("ParentProcess", {}).get("ImageFile", {}).get("Name")):
            parent_process_names.append(entity["ParentProcess"]["ImageFile"]["Name"])
    
    return parent_process_names
```

cc @RomuDeuxfois